Earthfile

```diff
@@ -47,9 +47,14 @@ generate:
 # e2e runs end-to-end tests. See test/e2e/README.md for details.
 e2e:
   ARG FLAGS="-test-suite=base"
-  # Docker installs faster on Alpine, and we only need Go for go tool test2json.
-  FROM golang:${GO_VERSION}-alpine3.20
-  RUN apk add --no-cache docker jq
+  # Using earthly image to allow compatibility with different development environments e.g. WSL
+  FROM earthly/dind:alpine-3.20-docker-26.1.5-r0
+  RUN wget https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz
+  RUN tar -C /usr/local -xzf go${GO_VERSION}.linux-amd64.tar.gz
```

Comment:
Amd64 only, I wonder if there would be a case where this could be an issue 🤔 can't we just install it with apk and let that handle the arch appropriately?

Reply:
Good point! Thanks to Docker's Rosetta project, AMD64 executables can run on ARM CPUs, so that code works on Docker on Mac.

It'll be complicated to rely on APK as if we just look at latest package version for Golang `1.23.4`, https://archlinux.org/packages/extra/x86_64/go/, its version is `2:1.23.4-1`, which is not 1:1 mapped to Golang. The prefix `2:` and suffix `-1` is arbitrary and potentially can break if any will change.

I updated the arch and OS, so it'll be (more) possible to run E2E tests on Podman and the code is more straightforward and future-proof.

Reply:
TIL: looks like you can do [fuzzy version matching](https://wiki.alpinelinux.org/wiki/Alpine_Package_Keeper#Holding_a_specific_package_back), so something like this should work: 
```
apk add go~=1.23.3
```

Reply:
Version matching would work without the prefix. It doesn't solve the issue I mentioned above. Applying this code with a prefix `2:`, throws below error:
```
                +e2e *failed* | --> RUN apk add --no-cache jq go~=2:${GO_VERSION}
                +e2e *failed* | fetch https://dl-cdn.alpinelinux.org/alpine/v3.20/main/aarch64/APKINDEX.tar.gz
                +e2e *failed* | fetch https://dl-cdn.alpinelinux.org/alpine/v3.20/community/aarch64/APKINDEX.tar.gz
                +e2e *failed* | ERROR: 'go~=2:1.23.4' is not a valid world dependency, format is name(@tag)([<>~=]version)
```

Without prefix, it tries to use latest version without `2:` prefix, version 1.22:
```
                +e2e | --> RUN echo ${GO_VERSION}
                +e2e | 1.23.4
                +e2e | --> RUN apk add --no-cache jq go~=${GO_VERSION}
                +e2e | fetch https://dl-cdn.alpinelinux.org/alpine/v3.20/main/aarch64/APKINDEX.tar.gz
                +e2e | fetch https://dl-cdn.alpinelinux.org/alpine/v3.20/community/aarch64/APKINDEX.tar.gz
                +e2e | ERROR: unable to select packages:
                +e2e |   go-1.22.10-r0:
                +e2e |     breaks: world[go~1.23.4]
```

That's why installing Golang manually is the only stable solution.

Reply:
Ah right, it's pulling from 3.20 of alpine, not edge 🤔 weird, it installed it properly when I tested it out. Then let's go with the manual install 🙏